packages/react/src/components/text-input/text-input.test.tsx

```diff
@@ -74,6 +74,15 @@ describe('TextInput', () => {
         expect(container.prop('valid')).toBe(true);
     });
 
+    test('should set as invalid when valid prop is false and input trigger blur again', () => {
+        const wrapper = shallow(<TextInput valid={false} />);
+
+        const textInput = getByTestId(wrapper, 'text-input');
+        textInput.simulate('blur', { currentTarget: { checkValidity: () => true } });
+
+        expect(wrapper.prop('valid')).toBe(false);
```

Comment:
Je pense que que le prop `valid` de `TextInput` devrait naturellement rester `false` et que c'est le `valid` de `field-container` qu'on devrait vérifier qu'il est false.

```suggestion
        const container = getByTestId(wrapper, 'field-container');
        expect(container.prop('valid')).toBe(true);
```

On pourrait aussi ajouter d'autres tests:
- Si  prop`valid` = `true` dans le cadre d'un blur: vérifier que le prop `valid` de field-container reste toujours égale à `true`.
- Les même tests mais avec `checkValidity` qui retourne false.

Reply:
J'ai rajouté les test et la validation est maintenant faite sur le field-container